### ISSUE TITLE
IF2 inflexibility in what the latent nodes are

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#935]
As written, IF2 allows a user to specify something like `alpha[2:20]` as the `nodes` even if 'alpha[1:100]` is defined in the model code, but then in the processing it decides that `alpha[1:100]` is the set of latent nodes. 

This could be confusing to the user and also prevents one from saying that `alpha[1]` is the baseline node. 

@perrydv let's discuss how much we should address this. For the moment I have put a cautionary note in the roxygen.

**Comment by paciorek** [comment]
Also @perrydv  for testing purposes, do you have any code lying around where the latent states are vectors?

**Comment by perrydv** [comment]
I think this is the issue that some of Nick's old code deals with, that is currently sitting on the side waiting to be worked back in.  I can point you to it if you want @paciorek.  No, I don't think I have testing code for vector latent states.

**Comment by paciorek** [comment]
Yes please point me to that code.

**Comment by perrydv** [comment]
I think it is in branch pfLatentSpecificationMods (and this is the only remaining piece we need from that branch).  The issue is relevant for any of the filtering algos, and this branch is pre-IF2, so look in the other filtering build*.R nimbleFunctions, e.g. buildAuxiliaryFilter.R.  There's some thick grepping  on the node names, so it might a bit of a dive to follow it.

**Comment by paciorek** [comment]
Ok, I have code in a function called 'findLatentNodes' that modifies Nick's functionality (replacing grepping with manipulation of expressions) to handle cases like:

```
nodes = 'x'
nodes = 'x[2:4, 3:20]'
nodes = c('x[1]','x[2]',....,'x[20]')
nodes = c('x[,1]','x[,2]',....,'x[,20]')
```

It fills in missing indexes, determines time dimension and then expands so that there is one 'node' per time point (it might actually be a set of nodes at each time point).

Will embed use of this function in all PF algos and will modify roxygen for 'nodes' to make clear what specification is allowed.

**Comment by paciorek** [comment]
Fixed in PR #940
